lib/xnvmec.c

```diff
@@ -1561,13 +1561,13 @@ xnvmec_assign_arg(struct xnvmec *cli, struct xnvmec_opt_attr *opt_attr, char *ar
 		args->nsr = num;
 		break;
 	case XNVMEC_OPT_AD:
-		args->ad = arg ? num : 0;
+		args->ad = arg ? num : 1;
```

Comment:
Generally, then the values parsed from the cli are 0 by default.
This would flip, such that arguments which are not provided will carry the value 1 by default.

In case it should be restricted to the set of values 1 or 0, then how about changing this to:
```
args->idw = arg ? 1 : 0;
```

Or, am I missing something here?

Reply:
I followed what was done for XNVMEC_OPT_HELP which is also one that does not carry an argument. Should we change that one to `arg ? 1 : 0` as well then?

Reply:
Hmm, I need to take another look, seems like there is something I have forgotten about the cli-parser code here.
As I recall, then point of ``args->attribute = arg ? num : k;`` is that the logic can be used for arguments which are used as either value-options or flag-options. When used as flags, then I have forgotten why there is a discrepancy in the code-base of using:

```
args->ad = arg ? num : 1;
```
in some cases, and:
```
args->ad = arg ? num : 0;
```
In others... I would expect the latter form to be "correct", that is, default to 0.

Reply:
One approach here, is make the change that works for you for the ``dsm`` functionality, these are only available on ``next`` not in any release. Thus, their behavior will not break anything. Then, I or you could write an issue about the need to investigate this and cleanup/refactor the code for proper common operation.

Reply:
What I mean by this is:

* Add the cli-parsing that you need
* Don't change the behavior of the other arguments (to avoid regression)
* Open an issue, to clean the argument parser "up" / re-factor

Reply:
Just gave the code a quick run and here is what I understand is happening:
1. With XNVMEC_LFLG type arguments arg is 0x0 (because there is no accompanying  arg) and num is always 0 because the code that sets it is skipped in line 1207.
2. Every time you are in a case of a XNVMEC_LFLG type arg you should just consider it being in the arguments

IMO the refactoring should be changing all the XNVMEC_LFLG type args to args->ARG_NAME = 1; or whatever should happen when the arg is present in the command line.

Reply:
> What I mean by this is:
> 
> * Add the cli-parsing that you need
OK

> * Don't change the behavior of the other arguments (to avoid regression)
Do you mean IDW and IDR here?  I'm ok with that, but they will be broken if you don't do anything.

> * Open an issue, to clean the argument parser "up" / re-factor
I can do that know if you want.